project/src/main/webapp/shared.js

```diff
@@ -55,31 +82,124 @@ async function moveTagsFromPoolToInput(clickedAdder, tagId) {
 
 /** Add tag to user input box once it is chosen */
 function addTagsToInput(clickedAdder, tagId) {
-  $.getScript('https://cdnjs.cloudflare.com/ajax/libs/bootstrap-tagsinput/0.8.0/bootstrap-tagsinput.js', function() {
-    $(tagId).on('itemAdded', function() {
-      if ($(tagId).prevAll().length > 2) {
-        const otherId = tagId === '#interests' ? '#skills' : '#interests';
-        $(otherId).prev().remove();
-        $(tagId).prev().prev().remove();
-      }
-    });
-    $(tagId).on('itemRemoved', function(event) {
-      addTagBackToPool(event.item, tagId);
-    });
+  $(tagId).on('itemAdded', function() {
+    if ($(tagId).prevAll().length > 2) {
+      const otherId = tagId === '#interests' ? '#skills' : '#interests';
+      $(otherId).prev().remove();
+      $(tagId).prev().prev().remove();
+    }
+  });
+  $(tagId).on('itemRemoved', function(event) {
+    addTagBackToPool(event.item, tagId);
+  });
+  getTagsScriptWithCallback(function() {
     $(tagId).tagsinput('refresh');
     $(tagId).tagsinput('add', $(clickedAdder).next().html());
   });
 }
 
+function getTagsScriptWithCallback(fn) {
```

Comment:
maybe add javadoc, I think it is unclear what fn means.

Reply:
Done, renamed the argument to `callback` to avoid a doc